### ISSUE TITLE
Missing meta fields for query/type

### ISSUE THREAD

**Issue by jakubknejzlik** [nautilus/gateway#72]
According to GraphQL specs, it should be possible to query meta fields (query's `__schema` and `__type` and type's `__typename` - not sure if that's all).

Currently the gateway doesn't expose them, which results in following error while using apollo client:
```
GraphQL error: Could not find location for XXX.__typename
```

I'ts possible to disable sending `__typename` field using
```
export const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache({
    addTypename: false
  })
});
```
but this isn't always an option.

**Comment by AlecAivazis** [comment]
Hey @jakubknejzlik!

Thanks for another issue report. I am pretty sure that those fields do exist since the playground can introspect them. Also, I think the error would look more like `can't query field __typename on {Type}`. I suspect this is because I don't populate `Location` correctly somewhere 🤔

**Comment by AlecAivazis** [comment]
Do you by chance have a repo that reproduces this?

**Comment by jakubknejzlik** [comment]
@AlecAivazis not a repo, but you can try latest gateway version with this api:
https://graphql-orm-example.herokuapp.com/graphql

and query(example):
```
query {
  company {
    __typename
  }
}
```

Direct response:
```
{
  "data": {
    "company": {
      "__typename": "Company"
    }
  }
}
```

Behind gateway:
```
{
  "data": {},
  "errors": [
    {
      "extensions": {
        "code": ""
      },
      "message": "Could not find location for Company.__typename"
    }
  ]
}
```

**Comment by AlecAivazis** [comment]
Ah!! Sorry this error message is really bad. This isn't not about the Location that is part of the introspection query. This error is the gateway saying that it could not find a service that defined `__typename` on the `Company` type. I should have a fix for this soon.

**Comment by jakubknejzlik** [comment]
Hmm...interesting. I've found multiple GraphQL API's, which doesn't expose __typename field in schema, but correctly resolve it.
For example: https://gitlab.com/api/graphql

Unfortunately trying to run the gateway with:
```
./gateway start --port 4000 --services https://gitlab.com/api/graphql
```
returns the error: ```Encountered error introspecting schemas: No operation named ""```
But it's different story :)

**Comment by jakubknejzlik** [comment]
This is working example:
```
./gateway start --port 4000 --services https://countries.trevorblades.com/
```

Query:
```
query {
  continents {
    __typename
    code
    name
  }
}
```

Isn't the problem that the metadata fields are resolved, but not specified specifically in schema? 🤔

**Comment by jakubknejzlik** [comment]
Ah, sorry for spam...I misunderstood your comment...I think it's clear by looking at the PR 👍

**Comment by AlecAivazis** [comment]
Yea I noticed that too 🤔 I think graphql-js doesn't include it in the introspection results but the playground allows it to be added (just a random guess). Anyway, I think I fixed this problem. There is a PR waiting for your approval 😄 

When the gateway is merging the service schemas together, it has to keep track of where each field is defined. For reasons that don't matter `__typename` was not included in that list so the gateway could not find a location to find the field definition
